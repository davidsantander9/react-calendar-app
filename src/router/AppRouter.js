import React, { useEffect } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Redirect
  } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { startCheckin } from '../actions/auth';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {

    const dispatch = useDispatch();

    const { checking, uid } = useSelector( state => state.auth );
    const uidCheck = uid === undefined ? null : uid

    useEffect(() => {
        dispatch( startCheckin() );
    }, [dispatch]);

    if ( checking ) {
        return (<h5>Wait...</h5>)
    } 

    return (
        <Router>
            <div>
            <Switch>
                    <PublicRoute
                        exact 
                        path='/login' 
                        component={ LoginScreen }
                        isAuthenticated={ !!uidCheck }
                        />
                    <PrivateRoute 
                        exact 
                        isAuthenticated={ !!uidCheck }
                        path='/' 
                        component={ CalendarScreen }/>
                    <Redirect to='/' />
                </Switch>
            </div>
        </Router>
    )
}
