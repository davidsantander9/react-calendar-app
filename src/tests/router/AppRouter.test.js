import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { AppRouter } from '../../router/AppRouter';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );



let store = mockStore( initState );
// store.dispatch = jest.fn();



describe('Test on <AppRouter/>', () => {

    const initState = {
        auth: {
            checking: true
        }
    };

    
    test('should show wait', () => {
        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter/>
            </Provider>
        );
        

        expect( wrapper ).toMatchSnapshot();
    });
    

})
