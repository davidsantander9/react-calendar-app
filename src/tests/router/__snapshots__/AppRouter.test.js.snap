// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter/> should show wait 1`] = `
<h5>
  Wait...
</h5>
`;
